src/store/TimeperiodStore.js

```diff
@@ -19,13 +19,59 @@ export default class TimeperiodStore {
 
     @observable timeUnit = null;
     @observable interval = null;
+    @observable remain = "00:00";
 
     onContextReady = () => {
         const { timeUnit, interval } = this.context.stx.layout;
         this.timeUnit = getTimeUnit({ timeUnit, interval });
         this.interval = interval;
+
+        const stx = this.context.stx;
+
+
+
+        reaction(() => this.timeUnit , () => { this.showCandleCountdown(); });
+
+        const displayMilliseconds = (ms) => {
+            const totalSec = ms / 1000;
+            if (totalSec <= 0) { return null; }
+            const padNum = (n) => (`0${n}`).slice(-2);
+            const seconds = padNum(Math.trunc((totalSec) % 60));
+            const minutes = padNum(Math.trunc((totalSec/60) % 60));
+            let hours = Math.trunc((totalSec / 3600) % 24);
+            hours = hours ? `${hours}:` : '';
+            return `${hours}${minutes}:${seconds}`;
+        };
+
+        setInterval(() => {
+            var dataSet = stx.chart.dataSet;
+            if(dataSet && dataSet.length != 0 ){
+                let diff = new Date() - dataSet[dataSet.length-1].DT;
+                this.remain = displayMilliseconds((getIntervalInSeconds(stx.layout) * 1000) - diff);
+                this.remain = this.remain ? this.remain : "00:00";
+                this.showCandleCountdown();
+            }
+        }, 1000);
```

Comment:
I'm not sure if we should update more often than 1 second. There might be a couple millisecond delay or smth. Maybe we could make it 500 milliseconds instead? If 1 second works fine then sure

Reply:
I think we'd better keep it to 1 second, i see better result with 1 sec interval and i have a slow internet :)  plus when i make it 500 milliseconds, it reduce the time for 2 sec each time like its 10:58 to 10:56, i dont why exactly that happen maybe calling it too times cause a delay, but we'd better stay with 1 sec.

Reply:
aite then. :+1: